python/housinginsights/sources/base.py

```diff
@@ -164,7 +165,95 @@ def directly_to_file(self, data, filepath):
+        id by performing a lookup on the project table in the database using
+        the mar_id column.
+
+        If the address id doesn't map to an existing building in the table,
+        a randomly generated uuid is returned as nlihc_id.
+        """
+        query = "select nlihc_id from project where mar_id = " \
+                "'{}';".format(address_id)
+        query_result = db_conn.execute(query)
+        result = [dict(x) for x in query_result.fetchall()]
+
+        if result:
+            return result[0]['nlihc_id'], True
+        else:
+            return str(uuid4()), False
 
+    def create_project_subsidy_csv(self, uid, project_fields_map,
+                                   subsidy_fields_map, database_choice=None):
+        """
+        Writes 'source_project' and 'source_subsidy' raw files from the
+        source csv file. It then deletes the source file so it
+        doesn't get added to manifest and loaded into the database.
+        """
+        if database_choice is None:
+            database_choice = 'docker_database'
+        engine = dbtools.get_database_engine(database_choice)
+        db_conn = engine.connect()
+
+        # create file path objects
+        source_csv = self.output_paths[uid]
+        folder = os.path.dirname(source_csv)
+        source_proj = os.path.join(folder, "{}_project.csv".format(uid))
+        source_subsidy = os.path.join(folder, "{}_subsidy.csv".format(uid))
+
+        # create dchousing_project/subsidy files from source
+        with open(source_csv, encoding='utf-8') as f, \
+                open(source_proj, mode='w', encoding='utf-8') as proj, \
+                open(source_subsidy, mode='w', encoding='utf-8') as subsidy:
+
+            reader = csv.DictReader(f)
+            proj_writer = csv.DictWriter(proj, fieldnames=PROJ_FIELDS)
+            proj_writer.writeheader()
+            subsidy_writer = csv.DictWriter(subsidy, fieldnames=SUBSIDY_FIELDS)
+            subsidy_writer.writeheader()
+
+            for line in reader:
+                nlihc_id, in_proj_table = self._get_nlihc_id_from_db(
+                    db_conn=db_conn, address_id=line['ADDRESS_ID'])
+
+                if not in_proj_table:
+                    data = self._map_data_for_row(nlihc_id=nlihc_id,
+                                                  fields=PROJ_FIELDS,
+                                                  fields_map=project_fields_map,
+                                                  line=line)
+                    proj_writer.writerow(data)
+
+                data = self._map_data_for_row(nlihc_id=nlihc_id,
+                                              fields=SUBSIDY_FIELDS,
+                                              fields_map=subsidy_fields_map,
+                                              line=line)
+                subsidy_writer.writerow(data)
+
+        # remove source file so it doesn't get added to manifest hence database
```

Comment:
Clarify why you'd want to remove the 'source' file. 

"source_csv is the data we are trying to merge with our existing project and subsidy tables. Since we have the data split into appropriate proj_ and subsidy_ csv files we no longer need the source_csv file."

Long term I might consider figuring out a way for the source_csv to be excluded from the merge but still not deleted so we can trace down problems if needed, but no big deal for now.